src/main/scala/com/starrocks/connector/spark/catalog/StarRocksCatalog.scala

```diff
@@ -0,0 +1,102 @@
+
+class StarRocksCatalog extends TableCatalog
+  with SupportsNamespaces
+  with FunctionCatalog
+  with Logging {
+
+  private var catalogName: String = _
+  private var properties: util.Map[String, String] = _
+  private var simpleStarRocksConfig: SimpleStarRocksConfig = _
+
+  override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {
+    this.catalogName = name
+    this.properties = addPrefixInStarRockConfig(options.asCaseSensitiveMap())
+    this.simpleStarRocksConfig = new SimpleStarRocksConfig(properties)
+
+    log.info(s"StarRocks clusters' detail: $options")
+  }
+
+  override def name(): String = catalogName
+
+  override def listTables(namespace: Array[String]): Array[Identifier] = {
+    val table2Db = StarRocksConnector.getTables(simpleStarRocksConfig, namespace.toList.asJava)
+    table2Db.map { case (tb, db) => Identifier.of(Array(db), tb) }.toArray
+  }
+
+  override def loadTable(ident: Identifier): Table = {
+    val starRocksSchema = getStarRocksSchema(simpleStarRocksConfig, ident)
+    val schema = InferSchema.inferSchema(starRocksSchema, simpleStarRocksConfig)
+    new StarRocksTable(schema, starRocksSchema, simpleStarRocksConfig, ident)
+  }
+
+  override def createTable(ident: Identifier,
+                           schema: StructType,
+                           partitions: Array[Transform],
+                           properties: util.Map[String, String]): Table = throw new UnsupportedOperationException
+
+  override def alterTable(ident: Identifier, changes: TableChange*): Table = throw new UnsupportedOperationException
+
+  override def dropTable(ident: Identifier): Boolean = throw new UnsupportedOperationException
+
+  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = throw new UnsupportedOperationException
+
+  override def listNamespaces(): Array[Array[String]] = {
+    val dbNames = StarRocksConnector.getDatabases(simpleStarRocksConfig)
+    dbNames.map(dbName => Array(dbName)).toArray
+  }
+
+  override def listNamespaces(namespace: Array[String]): Array[Array[String]] = throw new UnsupportedOperationException
+
+  override def loadNamespaceMetadata(namespace: Array[String]): util.Map[String, String] = {
+    StarRocksConnector.loadDatabase(simpleStarRocksConfig, namespace.toList.asJava)
+  }
+
+  override def createNamespace(namespace: Array[String], metadata: util.Map[String, String]): Unit =
+    throw new UnsupportedOperationException
+
+  override def alterNamespace(namespace: Array[String], changes: NamespaceChange*): Unit =
+    throw new UnsupportedOperationException
+
+  override def dropNamespace(namespace: Array[String], cascade: Boolean): Boolean =
+    throw new UnsupportedOperationException
+
+  override def listFunctions(namespace: Array[String]): Array[Identifier] = throw new UnsupportedOperationException
+
+  override def loadFunction(ident: Identifier): UnboundFunction = throw new UnsupportedOperationException
```

Comment:
If not support function methods, should not inherit `FunctionCatalog`?

Reply:
ok,